site/ja/guide/create_op.md

```diff
@@ -0,0 +1,1231 @@
+    .Input("in: T")
+    .Output("out: T");
+
+REGISTER_OP("RestrictedTensorSequenceExample")
+    .Attr("T: list({int32, int64})")
+    .Input("in: T")
+    .Output("out: T");
+```
+
+両方ともおなじ型 `T` であるため、出力 `out` のテンソルの数と型は、入力 `in` とおなじであることに注意してください。
+
+* おなじ型をもつ連続したテンソルについて: `<number> * <type>` となる。`<number>` は、`int` 型である [アトリビュート](#アトリビュート) の名前である。`<type>` は、`tf.DType` または `type` 型のアトリビュートの名前である。最初の例では、オペレーションは `int32` のテンソルのリストを受け取る。
+
+```c++
+REGISTER_OP("Int32SequenceExample")
+    .Attr("NumTensors: int")
+    .Input("in: NumTensors * int32")
+```
+
+一方このオペレーションは、すべておなじである限り、いかなる型のテンソルのリストを受け取ります。
+
+```c++
+REGISTER_OP("SameTypeSequenceExample")
+    .Attr("NumTensors: int")
+    .Attr("T: type")
+    .Input("in: NumTensors * T")
+```
+
+* テンソルのリファレンスついて: `Ref(<type>)` となる。`<type>` は前述した型のいずれかになる。
+
+> 命名に関する注意点: 入力の型として利用されている、いかなるアトリビュートは推測されます。慣例的に、推測されたアトリビュートは、（`T` や `N` のように）大文字の名前を使用します。そうでなければ、入力や出力、アトリビュートは、関数のパラメータ（たとえば `num_outputs`）のような名前を持ちます。詳細は、[前述した命名の注意点](#naming) を参照してください。
+
+詳細は、[`tensorflow/core/framework/op_def_builder.h`][op_def_builder] を参照してください。
+
+
+#### 後方互換性
+
+すばらしいカスタムオペレーションを作り、ほかの人と共有し、顧客がそのオペレーションを使って喜んでいることを想定しましょう。
+しかし、何らかの方法でオペレーションを変更したいとします。
+
+一般的に、既存のチェックインされた仕様の変更は、後方互換でなければなりません。
+すなわち、オペレーションの仕様変更は、前に古い仕様から構築してシリアライズ化された `GraphDef` プロトコルバッファを壊してはいけません。
+`GraphDe` の互換性の詳細は、[ここで説明されています](./versions.md#compatibility_of_graphs_and_checkpoints)。
+
+後方互換性を保つための方法は、いくつかあります。
+
+1. オペレーションに新しく追加されたアトリビュートは、デフォルト値が定義されている必要があり、そのデフォルト値の場合には、オペレーションはもともとの動作にならなければなりません。非ポリモーフィズムからポリモーフィズムに変更するためには、新しい型アトリビュートにデフォルト値を与え、デフォルトでもとのシグネイチャを保つ必要があります。たとえば、オペレーションが、次のようなものであった場合、
+
+       REGISTER_OP("MyGeneralUnaryOp")
+           .Input("in: float")
+           .Output("out: float");
+
+   次のようにして、後方互換を維持しながらポリモーフィズムにできます。
+
+       REGISTER_OP("MyGeneralUnaryOp")
+           .Input("in: T")
+           .Output("out: T")
+           .Attr("T: numerictype = DT_FLOAT");
+
+2. アトリビュートの制約を緩めるのは安全に行えます。たとえば、`{int32, int64}` から `{int32, int64, float}` もしくは `type` に変更できます。また、`{"apple", "orange"}` を `{"apple", "banana", "orange"}` または `string` に変更できます。
+
+3. リストの型のデフォルトが古いシグネイチャと一致する限り、単一の入力/出力をリストの入力/出力に変更できます。
+
+4. デフォルトが空であれば、新たなリストの入力/出力を追加できます。
+
+5. オペレーションの名前に、プロジェクトに固有の名前をプレフィックスすることで、作成したオペレーションの名前空間を作れます。これにより、TensorFlowの将来のバージョンで含まれるかもしれないオペレーションとの衝突を回避できます。
+
+6. 前もって計画してください！オペレーションが使われる将来を予想しましょう。いくつかのシグネイチャの変更は、互換を保つ方法ではできません（たとえば、おなじ型のリストを異なる型のリストに変更するなど）。
+
+変更が安全か安全でないかの一覧は、[`tensorflow/core/framework/op_compatibility_test.cc`](https://www.tensorflow.org/code/tensorflow/core/framework/op_compatibility_test.cc) にあります。
+もし、オペレーションの変更を後方互換にできないならば、新しい名前かつ新しいセマンティクスで、新しいオペレーションを作ります。
+
+これらの変更は、`GraphDef` の互換性を維持できますが、生成されたPythonコードは、呼び出し側にとって互換にならない変更をもたらすかもしれません。
+Python APIは、新しい任意の引数を最後に追加するなどして古いシグネイチャを保つなど、手書きのPythonラッパー内で注意深く変更することで、互換性を保てるかもしれません。
+一般的に、互換性のない変更は、TensorFlowのメジャーバージョンが変更されたときのみ許されており、[`GraphDef` のバージョンのセマンティクス](./versions.md#compatibility_of_graphs_and_checkpoints) に従う必要があります。
```

Comment:
Should we use the full path to point to `versions.md` of the English version?

Reply:
Yes, we should. I fixed it to point to English version.